action.yml

```diff
@@ -32,12 +32,15 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Install PyYAML
+    - name: Setup python venv and dependencies
       shell: bash
-      run: pip install --user PyYAML
+      run: |
+        python3 -m venv .venv
+        source .venv/bin/activate
+        python3 -m pip install PyYAML
 
     - name: Run Clifus
-      shell: bash
+      shell: bash -e -l {0}
```

Comment:
does -l helps keeping the venv active?

Reply:
From what i understood (from https://github.com/orgs/community/discussions/25061) yes, it sources the profile of the first step.

Reply:
Why not just merge the two steps?

Reply:
That would be too simple and i don't like simple things

Reply:
Also there are other steps using the output from the clifus step

Reply:
We could do it in one step, but it would require even more refactoring. I'll do it if this fix fails to fix the problem.

Reply:
This will not work profile is not update by pip tooling

Reply:
You have no other choice than:
- merging steps
Or
- rerunning ˋsource venv\bin\activateˋ